apps/api/src/app/layouts/use-cases/set-default-layout/set-default-layout.use-case.ts

```diff
@@ -0,0 +1,68 @@
+import { LayoutEntity, LayoutRepository } from '@novu/dal';
+import { Injectable, Logger } from '@nestjs/common';
+
+import { SetDefaultLayoutCommand } from './set-default-layout.command';
+
+import { LayoutDto } from '../../dtos/layout.dto';
+import { EnvironmentId, LayoutId, OrganizationId } from '../../types';
+
+@Injectable()
+export class SetDefaultLayoutUseCase {
+  constructor(private layoutRepository: LayoutRepository) {}
+
+  async execute(command: SetDefaultLayoutCommand) {
+    const defaultLayoutId = await this.findDefaultLayoutId(command.environmentId, command.organizationId);
+
+    try {
+      if (defaultLayoutId) {
+        await this.setIsDefaultForLayout(defaultLayoutId, command.environmentId, command.organizationId, false);
+      }
+
+      await this.setIsDefaultForLayout(command.layoutId, command.environmentId, command.organizationId, true);
+    } catch (error) {
+      Logger.error(error);
+      // TODO: Rollback through transactions
```

Comment:
We have no transactions implementation in the Mongoose adapter repository. Something to do across the stack.